### ISSUE TITLE
negative sampling and loss

### ISSUE THREAD

**Issue by XiaoLiuAI** [dhwajraj/deep-siamese-text-similarity#24]
As I understand, you sampled double negative examples and treated all the examples equally in the loss. Would it hurt the performance if I increase the negative samples? Because the loss will become very small even if the model predict all examples as negative examples when I, for example, sampled 64 times negative examples than positive ones.

**Comment by dhwajraj** [comment]
yes it would hurt the performance if you increase the negative samples a lot. you can keep the 1:2 or max 1:3 ratio, more than that would bias the model.

**Comment by XiaoLiuAI** [comment]
Is there any way to avoid this? I remember that negative sampling ratio used in language model in tensorflow tutorial is 64, but I don't understand the details of the loss function they used. 

In addition, why you choose not to share variables between two bi-LSTMs? I tried an experiment on a practical data set and I found many times the model did not give the minimum distance for the pair of identical texts. If you want to model directional relation like is_a, shouldn't it be a single bi-LSTM plus a mapping matrix M, so text1, text2 are encoded into Vec1, Vec2 and Vec2 can be obtained by Vec1 x M? Like in https://aclweb.org/anthology/P/P14/P14-1113.xhtml.

**Comment by dhwajraj** [comment]
Not sure how that would perform. I wanted to try encode-decoder architecture as well. Will definitely look into these approaches but some time later. 
If you get time , can you please run some prototype experiments to contribute in this project.
